tmt/steps/provision/vagrant.py

```diff
@@ -248,64 +386,84 @@ def plugin_install(self, name):
+
+        self.debug('provision dir', self.provision_dir)
+
+         # Check for working Vagrant
+        self.run_vagrant('version')
+
+        # if custom vagrantfile was provided
+        if not self.custom_vagrantfile is None \
+            and self.custom_vagrantfile != self.vagrantfile \
+            and os.path.exists(self.custom_vagrantfile):
+            shutil.copy(self.custom_vagrantfile, self.vagrantfile)
+
+        # Don't setup Vagrantfile in case we are resuming, or custom
+        # Vagrantfile was specified,
+        if os.path.exists(self.vagrantfile) and os.path.isfile(self.vagrantfile):
+            self.validate()
+            return
+
+        # Let's add what's needed
+        # Important: run this first to install provider
+        self.add_how()
+
+        # Add default entries to Vagrantfile
+        self.add_defaults()
 
-    ## Knowhow ##
     def check_input(self):
-        """ Initialize configuration(sets defaults), based on data (how, image).
+        """ Initialize configuration(no defaults), based on data (how, image).
             does not create Vagrantfile or add anything into it.
         """
         self.debug('VagrantProvider', 'Checking initial status, setting defaults.')
 
-        self.set_default('how', 'virtual')
-        self.set_default('image', self.default_image)
-
-        image = self.data['image']
-
-        if self.is_uri(image):
-            self.set_default('box', 'box_' + self.instance_name)
-
-            if re.search(r"\.box$", image) is None:
+        if self.is_uri(self.image):
+            if re.search(r"\.box$", self.image) is None:
                 # an actual box file, Great!
                 pass
 
-            elif re.search(r"\.qcow2$", image) is None:
+            elif re.search(r"\.qcow2$", self.image) is None:
                 # do some qcow2 magic
-                self.data['box'] = '...'
+                self.box = '...'
                 raise SpecificationError("NYI: QCOW2 image")
 
             else:
-                raise SpecificationError(f"Image format not recognized: {image}")
+                raise SpecificationError(f"Image format not recognized: {self.image}")
 
         else:
-            self.set_default('box', image)
-            self.data['image'] = None
-
-        self.set_default('memory', self.default_memory)
-
-        # General ssh config, used for 'managed' as well
-        self.set_default('user', self.default_user)
+            if self.box is None:
+                self.box  = self.image
+                self.image = None
 
-        for key, val in self.data.items():
-            if self.debugon or key in self.display:
-                if not val is None:
-                    self.info(f'{key}', val)
 
+    ## Knowhow ##
     def add_how(self):
         """ Add provider (in Vagrant-speak) specifics """
         getattr(self,
-            f"how_{self.data['how']}",
+            f"how_{self.parent.how}",
             self.how_generic,
             )()
```

Comment:
Could be one line?

Reply:
Sure. I simply found it enhanced readability to have it multi-line.